### PR TITLE
6601 load plugin manifest

### PR DESCRIPTION
**Pull request by tsmaeder**
#### What it does

Fixes "Frontend plugins are broken" #6601 by loading the package.json upon initializing the plugin.

Note: I haven't been able to make the electron case work, since I don't know how to compute the proper URI to load package.json. However, since I'm going to be on vacation, I wanted to propose this PR anyway...maybe someone can fix it.

#### How to test
Build a front end plugin (for example https://github.com/eclipse/che-theia-samples/tree/master/samples/hello-world-frontend-plugin) and add the resulting plugin to the theia plugins folder.
Start Theia and ensure that the "Hello World" command has been added via `F1`. 

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by marechal-p — COMMENTED** [review]
I get an error when running the vscode ext you suggested:

> root ERROR Could not transfer 'package.nls.json' file from 'che__eclipse_che_hello_world_frontend_plugin' Error: ENOENT: no such file or directory, stat 'c:\Users\me\AppData\Local\Temp\theia-unpacked\eclipse_che_hello_world_frontend_plugin.theia\package.nls.json'

The ext is indeed missing a `package.nls.json` file. Is it a mandatory file?

**Comment by benoitf** [comment]
I think it's not handling the case where a file could be optional as package.nls.json may be missing

**Comment by tsmaeder** [comment]
> 
> 
> I get an error when running the vscode ext you suggested:
> 
> > root ERROR Could not transfer 'package.nls.json' file from 'che__eclipse_che_hello_world_frontend_plugin' Error: ENOENT: no such file or directory, stat 'c:\Users\me\AppData\Local\Temp\theia-unpacked\eclipse_che_hello_world_frontend_plugin.theia\package.nls.json'
> 
> The ext is indeed missing a `package.nls.json` file. Is it a mandatory file?

No, the error should be handled up the call chain, though.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (tsmaeder) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:36]: If all you know is a hammer ;-) No, there's no reason.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (tsmaeder) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:65]: Copy paste from the other "plugin-manifest-loader.ts". No Idea where that is coming from.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (marechal-p) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:36]: I'd prefer using `fetch`, but I'll leave it up to you since ultimately it works.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (tsmaeder) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:65]: @akosyakov any idea what the correct link might be here?

**Comment by tsmaeder** [comment]
@marechal-p  are you using electron? The description states that I have no idea how to compute the URI in this case:

> Note: I haven't been able to make the electron case work, since I don't know how to compute the proper URI to load package.json. 

If you have any suggestions, I'm all ears. Otherwise, since front-end plugins don't work at all, I think we should still merge the PR, it makes front-end plugins at least work in the browser case.

**Review by marechal-p — APPROVED** [review]
@tsmaeder I was convinced I had tried on the browser version, but you might be right. Tested again using browser and it works well. Electron does not.

LGTM.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (akosyakov) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:65]: cc @marcdumais-work

Comment (tsmaeder) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts]: Since we can't seem to find the proper uri I've removed the link.

Comment (benoitf) [reply packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts]: I think link is for https://github.com/theia-ide/vscode-builtin-extensions/blob/2997b3e9b2522e48ece117159f88af02b5f8dc7a/src/publish.js#L61

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (marechal-p) [inline packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:2]: 2020?

**Review by marechal-p — APPROVED** [review]

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (vince-fugnitto) [inline packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:2]: minor: should it be 2020?

**Comment by vince-fugnitto** [comment]
@tsmaeder you can restart the build if you like, sometimes we clear the Travis `cache` and it helps.

**Comment by tsmaeder** [comment]
I tried restarting the build, but it had no effect. Where do I click?

**Comment by vince-fugnitto** [comment]
> I tried restarting the build, but it had no effect. Where do I click?

I'll try for you :)

**Comment by tsmaeder** [comment]
I don't get the "rerun" buttons in the rightmost columns. Who do I need to bribe?

**Comment by vince-fugnitto** [comment]
> I tried restarting the build, but it had no effect. Where do I click?

I'm not sure :) @marcdumais-work maybe you know about how to give rights to maintainers for Travis?

**Comment by tsmaeder** [comment]
Now some other test fails and one of the platforms being tested actually passes. This is frustrating!

**Comment by tsmaeder** [comment]
Also, locally on Windows, I get another test failure (pre-hook for "move file" fails). Are the tests expected to be flaky?

**Comment by marcdumais-work** [comment]
> I don't get the "rerun" buttons in the rightmost columns. Who do I need to bribe?

You should normally be permitted to do this, since you have write-access to the repo. Can you access to the following?  https://travis-ci.com/github/eclipse-theia/theia/settings

**Comment by vince-fugnitto** [comment]
> Now some other test fails and one of the platforms being tested actually passes. This is frustrating!

I rebuilt, it previously failed since `open-vsx` was down and we could not download builtins for the tests.

File: packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts
Comment (marcdumais-work) [inline packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts:66]: This is to support using vscode built-ins from `npm` I think? Do you still use these in Che? 
We have a [couple of fixes](https://github.com/theia-ide/vscode-builtin-extensions/blob/master/src/package-vsix.js#L50-L71), to work-around problems with specific built-ins during vsix packaging - I do not think we do the same with the `npm` version.  

There is also [some extra massaging](https://github.com/theia-ide/vscode-builtin-extensions/blob/master/src/package-vsix.js#L113-L118) done  to each built-in's `package.json`, e.g. to make NLS work

**Comment by marcdumais-work** [comment]
> Tested again using browser and it works well. Electron does not.

Is this the case that because of the nature of this extension (it's Che-related after all), it's expected not to work on the Electron version of a Theia app?

**Comment by marechal-p** [comment]
@marcdumais-work more like that in a perfect world frontend extensions should work in any Theia product, but it was broken for both browser and electron contexts. This PR only fixes the browser part, and I don't know how to fix Electron either so hence why I am happy with the current changes.

**Comment by tsmaeder** [comment]
> 
> 
> > I don't get the "rerun" buttons in the rightmost columns. Who do I need to bribe?
> 
> You should normally be permitted to do this, since you have write-access to the repo. Can you access to the following? https://travis-ci.com/github/eclipse-theia/theia/settings

No, I get "Your permissions are insufficient".

**Comment by marcdumais-work** [comment]
> "Your permissions are insufficient"

I am not sure what the problem might be - the only hits on Google search seem to be CI for repos that their crawler was enable to access, giving this same error. 

One thing worth a try: On the following page, click on "sync account", on the upper left quadrant of the page: 
https://travis-ci.com/account/repositories

Do you see the "eclipse-theia" organization listed, under the button?

**Pull request merged**
